packages/taco/src/conditions/base/shared.ts

```diff
@@ -1,11 +1,19 @@
 import { z } from 'zod';
 
 import { ETH_ADDRESS_REGEXP, USER_ADDRESS_PARAM } from '../const';
+import { CONTEXT_PARAM_PREFIX } from '../context/context';
+
+export const ContextParamSchema = z.string().startsWith(CONTEXT_PARAM_PREFIX);
+export const ParamSchema = z.union([z.number(), z.string(), z.boolean()]);
+export const ParamOrContextParamSchema = z.union([
+  ParamSchema,
+  ContextParamSchema,
+]);
 
 export const returnValueTestSchema = z.object({
   index: z.number().optional(),
```

Comment:
To tighten this up (i.e. ensure index >= 0), can this be done:
```suggestion
  index: z.number().nonnegative().optional(),
```